### DISCUSSION TITLE
which python version recomemnd

### DISCUSSION THREAD

**Discussion by KaranaLJX** [AntonOsika/gpt-engineer#198] [Q&A] [answered]
i use python3.7 ，meet the following problem
➜  gpt-engineer git:(main) python3 main.py                
Traceback (most recent call last):
  File "main.py", line 6, in <module>
    from ai import AI
  File "/data00/home/linjiaxin.kara/python/gpt-engineer/ai.py", line 4, in <module>
    class AI:
  File "/data00/home/linjiaxin.kara/python/gpt-engineer/ai.py", line 30, in AI
    def next(self, messages: list[dict[str, str]], prompt=None):
TypeError: 'type' object is not subscriptable

**Comment by patillacode** [comment, marked as answer]
We are officially  compatible with python `3.8`, `3.9` and `3.10`

Can you try to run with any of those versions?

**Reply by KaranaLJX** [reply]
thanks，i have a try

**Comment by NoCLin** [comment]
@patillacode how about create executable by pyinstaller? I can do it

**Reply by patillacode** [reply]
I don't think we need that.

**Comment by TrevaRoe** [comment]
I think 3.10 is the best
